pkg/importer/dataStream_test.go

```diff
@@ -7,416 +7,404 @@ import (
+		table.Entry("expect failure trying to copy non-existing local image", filepath.Join(os.TempDir(), "cdi-testcopy"), "tinyCoreBad.iso", NewQEMUAllErrors(), true),
+		table.Entry("successfully copy streaming image", "cirros-qcow2.raw", "cirros-qcow2.img", NewFakeQEMUOperations(errors.New("should not be called"), nil, nil), false),
+		table.Entry("streaming image qemu validation fails", "cirros-qcow2.raw", "cirros-qcow2.img", NewFakeQEMUOperations(nil, nil, errors.New("invalid image")), true),
+		table.Entry("streaming image qemu convert fails", "cirros-qcow2.raw", "cirros-qcow2.img", NewFakeQEMUOperations(nil, errors.New("exit 1"), nil), true),
+	)
+
+	table.DescribeTable("internal copy", func(r io.Reader, out string, qemu bool, qemuOperations image.QEMUOperations, wantErr bool) {
+		defer os.Remove(out)
+		By("Replacing QEMU Operations")
+		replaceQEMUOperations(qemuOperations, func() {
+			err := copy(r, out, qemu)
+			if !wantErr {
+				Expect(err).NotTo(HaveOccurred())
+			} else {
+				Expect(err).To(HaveOccurred())
+			}
+		})
+	},
+		table.Entry("successfully copy reader", stringRdr, "testoutfile", false, NewFakeQEMUOperations(nil, errors.New("Shouldn't get this"), nil), false),
+		table.Entry("successfully copy qcow2 reader", testFileRdrs, "testqcow2file", true, NewFakeQEMUOperations(nil, errors.New("Shouldn't get this"), nil), false),
+		table.Entry("expect error trying to copy invalid format", testFileRdrs, "testinvalidfile", true, NewFakeQEMUOperations(nil, nil, errors.New("invalid format")), true),
+		table.Entry("expect error trying to copy qemu process fails", testFileRdrs, "testinvalidfile", true, NewFakeQEMUOperations(errors.New("exit 1"), nil, nil), true),
+	)
+})
+
+var _ = Describe("close readers", func() {
+	type args struct {
+		readers []reader
 	}
-	f()
-}
 
-func createDataStream(ep, accKey, secKey string) *DataStream {
-	dsurl, _ := ParseEndpoint(ep)
+	rdrs1 := ioutil.NopCloser(strings.NewReader("test data for reader 1"))
+	rdrs2 := ioutil.NopCloser(strings.NewReader("test data for reader 2"))
+	rdrA := reader{4, rdrs1}
+	rdrB := reader{7, rdrs2}
 
-	ds := &DataStream{
-		url:         dsurl,
-		buf:         make([]byte, image.MaxExpectedHdrSize),
-		accessKeyID: accKey,
-		secretKey:   secKey,
-	}
+	rdrsTest := []reader{rdrA, rdrB}
 
-	ds.constructReaders()
+	It("Should successfully close readers", func() {
+		err := closeReaders(rdrsTest)
+		Expect(err).NotTo(HaveOccurred())
+	})
+})
 
-	return ds
-}
+var _ = Describe("Random file name", func() {
+	const numbyte = 8
 
-func createDataStreamBytes(testfile, ep string, defaultBuf, singlereader bool) (*DataStream, []byte, error) {
-	urlPath := filepath.Join(ep, testfile)
-	if len(ep) == 0 {
-		urlPath = getURLPath(testfile)
-	}
-	testFilePath := getTestFilePath(testfile)
-	ds := createDataStream(urlPath, "", "")
+	It("create expected random name", func() {
```

Comment:
This test is really only testing that the random name is 8 bytes in length. It will always pass unless the length of random name used in `randTmpName` changes.

Reply:
Done